app/src/test/java/com/pmatuki/wowpapers/fake/WallpaperListSourceFake.kt

```diff
@@ -0,0 +1,37 @@
+package com.pmatuki.wowpapers.fake
+
+import com.pmatuki.wowpapers.data.remote.WallpaperListSource
+import com.pmatuki.wowpapers.model.Thumb
+import com.pmatuki.wowpapers.model.Wallpaper
+
+class WallpaperListSourceFake(val wallpaperCount: Int) : WallpaperListSource {
```

Comment:
Yes, that's the idea! 😊

In your fake class you can have some "setup functions" in order to test multiple behavior scenarios.
For example, you can have a "throwException" function that when you call "getWallpaperList()" will fail, you can have a "returnEmptyList" or else. 

Then your test will look something like:

```
@Test(expected = IllegalArgumentException::class)
fun `test if error is returned`()  = runBlockingTest {
    getWallpapers.throwError()
    val wallpaperList = getWallpapers()
}
```

You can do what you want with a Fake class and reuse in other tests without needing to mocking everything over and over again.